protocols/raft/src/main/java/io/atomix/protocols/raft/roles/LeaderRole.java

```diff
@@ -281,7 +281,14 @@ private void expireSession(RaftSessionContext session) {
           }
 
           log.trace("Appended {}", entry);
-          appender.appendEntries(entry.index());
+          appender.appendEntries(entry.index()).whenComplete((commitIndex, commitError) -> {
+            raft.checkThread();
+            if (isOpen()) {
```

Comment:
Can combine these two if statements :)

Reply:
Copying code always comes back to haunt me